source/extensions/common/wasm/context.cc

```diff
@@ -1125,6 +1125,28 @@ WasmResult Context::setProperty(std::string_view path, std::string_view value) {
   return WasmResult::Ok;
 }
 
+WasmResult Context::setEnvoyFilterState(std::string_view path, std::string_view value) {
+  auto* factory =
+      Registry::FactoryRegistry<StreamInfo::FilterState::ObjectFactory>::getFactory(path);
+  if (!factory) {
+    return WasmResult::NotFound;
+  }
+
+  auto object = factory->createFromBytes(value);
+  if (!object) {
+    return WasmResult::BadArgument;
+  }
+
+  auto* stream_info = getRequestStreamInfo();
+  if (!stream_info) {
+    return WasmResult::NotFound;
+  }
+
+  stream_info->filterState()->setData(path, std::move(object),
+                                      StreamInfo::FilterState::StateType::Mutable);
```

Comment:
Do we need flags to control lifespan? It is odd to set HTTP filter chain lifespan on TCP connections.

Reply:
I could pass the lifespan flag as part of protobuf if that makes sense? I wasn't sure if setting the lifespan was needed or not.

Reply:
I think it's needed since HTTP and network filters can use this API. Yes, adding to protobuf is fine, see https://github.com/envoyproxy/envoy/blob/707ceea7268ceacbe2b412bee689b3f950bbee0d/source/extensions/common/wasm/ext/declare_property.proto

Reply:
added the same `LifeSpan` flag and plumbed it through